iceoryx_posh/testing/include/iceoryx_posh/testing/mocks/posh_runtime_mock.hpp

```diff
@@ -45,6 +45,10 @@ class PoshRuntimeMock : public iox::runtime::PoshRuntime
         mockRuntime().reset();
     }
 
+#ifdef __clang__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
```

Comment:
As far as I can see it is to suppress complaints if variadic macros are invoked with no arguments. I suppose the Gtest `MOCK_METHOD` and so on are variadic and somehow called with zero arguments at times.

If so it is ok I guess since we want to use the Gtest macros.